src/shared/utilities/vsCodeUtils.ts

```diff
@@ -72,3 +73,59 @@ export async function closeAllEditors() {
         )
     }
 }
+
+/**
+ * Checks if the given extension is installed and active.
+ */
+export function isExtensionActive(extId: string) {
+    const extension = vscode.extensions.getExtension(extId)
+    return extension && extension.isActive
+}
+
+/**
+ * Activates the given extension, or does nothing if the extension is not
+ * installed.
+ *
+ * @param extId  Extension id
+ * @param silent  Return undefined on failure, instead of throwing
+ * @returns Extension object, or undefined on failure if `silent`
+ */
+export async function activateExtension(extId: string, silent: boolean = true): Promise<vscode.Extension<void> | undefined> {
+    let loggerInitialized: boolean
+    try {
+        getLogger()
+        loggerInitialized = true
+    } catch {
+        loggerInitialized = false
+    }
+    function log(s: string, ...rest: any[]): void {
+        if (loggerInitialized) {
```

Comment:
This is needed because getLogger() throws. Future enhancement: getLogger() could return a console.log impl instead of throwing.